js/pro/poloniex.js

```diff
@@ -0,0 +1,959 @@
+            const accessPath = '/ws';
+            const requestString = 'GET\n' + accessPath + '\nsignTimestamp=' + timestamp;
+            // let expires = this.milliseconds () + 10000;
+            // expires = expires.toString ();
+            const signature = this.hmac (this.encode (requestString), this.encode (this.secret), 'sha256', 'base64');
+            const request = {
+                'event': 'subscribe',
+                'channel': [ 'auth' ],
+                'params': {
+                    'key': this.apiKey,
+                    'signTimestamp': timestamp,
+                    'signature': signature,
+                    'signatureMethod': 'HmacSHA256',  // optional
+                    'signatureVersion': '2',          // optional
+                },
+            };
+            const message = this.extend (request, params);
+            future = await this.watch (url, messageHash, message);
+            //
+            //    {
+            //        "data": {
+            //            "success": true,
+            //            "ts": 1645597033915
+            //        },
+            //        "channel": "auth"
+            //    }
+            //
+            //    # Failure to return results
+            //
+            //    {
+            //        "data": {
+            //            "success": false,
+            //            "message": "Authentication failed!",
+            //            "ts": 1646276295075
+            //        },
+            //        "channel": "auth"
+            //    }
+            //
+            client.subscriptions[messageHash] = future;
+        }
+        return future;
+    }
+
+    async subscribe (name, isPrivate, symbols = undefined, params = {}) {
+        /**
+         * @ignore
+         * @method
+         * @description Connects to a websocket channel
+         * @param {String} name name of the channel
+         * @param {Bool} isPrivate true for the authenticated url, false for the public url
+         * @param {[String]|undefined} symbols CCXT market symbols
+         * @param {Object} params extra parameters specific to the poloniex api
+         * @returns {Object} data from the websocket stream
+         */
+        await this.loadMarkets ();
+        const publicOrPrivate = isPrivate ? 'private' : 'public';
+        const url = this.urls['api']['ws'][publicOrPrivate];
+        const subscribe = {
+            'event': 'subscribe',
+            'channel': [
+                name,
+            ],
+        };
+        const marketIds = [ ];
+        if (symbols !== undefined) {
+            for (let i = 0; i < symbols.length; i++) {
+                const symbol = symbols[i];
+                marketIds.push (this.marketId (symbol));
+            }
+        } else {
+            marketIds.push ('all');
+        }
+        if (name !== 'balances') {
+            subscribe['symbols'] = marketIds;
+        }
+        const messageHash = name + ':' + marketIds.join (',');
```

Comment:
This is going to cause issues if someone for examples calls passes a symbols argunment. For example: `watchTickers (['BTC/USDT', 'ETH/BTC']m)` the messageHash will be `tickers:BTC/USDT,ETH/BTC` which you won't be able to resolve the the handleTicker function